code/modules/reagents/chemistry/reagents/medicine_reagents.dm

```diff
@@ -109,23 +109,26 @@
 /datum/reagent/medicine/cryoxadone/on_mob_life(mob/living/M)
 	if(M.stat != DEAD && M.bodytemperature < T0C) // Low temperatures are required to take effect.
 		M.status_flags &= ~DISFIGURED
-		M.adjustCloneLoss(-1)
-		M.adjustOxyLoss(-5)
-		M.adjustBruteLoss(-1)
-		M.adjustFireLoss(-1)
-		M.adjustToxLoss(-1)
+		M.adjustCloneLoss(-2)
+		if(M.health < 1)
+			M.adjustOxyLoss(-5)
+			M.adjustBruteLoss(-1)
+			M.adjustFireLoss(-1)
+			M.adjustToxLoss(-1)
 	if(M.stat != DEAD && M.bodytemperature < 225) // At lower temperatures (cryo) the full effect is boosted
-		M.adjustCloneLoss(-1)
-		M.adjustOxyLoss(-2)
-		M.adjustBruteLoss(-2)
-		M.adjustFireLoss(-2)
-		M.adjustToxLoss(-2)
+		M.adjustCloneLoss(-3)
+		if(M.health < 1)
+			M.adjustOxyLoss(-5)
+			M.adjustBruteLoss(-2)
+			M.adjustFireLoss(-2)
+			M.adjustToxLoss(-2)
 	if(M.stat != DEAD && M.bodytemperature < 100) // At extreme temperatures (upgraded cryo) the effect is greatly increased.
```

Comment:
instead of checking stat and temperature three times, and calling each adjustxxxxx() proc three times, why not do:
```
if(stat != DEAD)
	switch(M.bodytemperature)
		if(0 to 100) blablabla
		if(100 to 225) derpderpderp
		if(225 to T0C) xxxxxx
```